pkg/runtime/processor/binding.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package processor
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/dapr/components-contrib/bindings"
+	"github.com/dapr/dapr/pkg/apis/common"
+	compapi "github.com/dapr/dapr/pkg/apis/components/v1alpha1"
+	compbindings "github.com/dapr/dapr/pkg/components/bindings"
+	diag "github.com/dapr/dapr/pkg/diagnostics"
+	"github.com/dapr/dapr/pkg/runtime/compstore"
+	rterrors "github.com/dapr/dapr/pkg/runtime/errors"
+	"github.com/dapr/dapr/pkg/runtime/meta"
+)
+
+const (
+	bindingTypeInput  = "input"
+	bindingTypeOutput = "output"
+	bindingDirection  = "direction"
+)
+
+type binding struct {
+	registry  *compbindings.Registry
+	compStore *compstore.ComponentStore
+	meta      *meta.Meta
+}
+
+func (b *binding) init(ctx context.Context, comp compapi.Component) error {
+	switch {
+	case b.registry.HasInputBinding(comp.Spec.Type, comp.Spec.Version):
+		if err := b.initInputBinding(ctx, comp); err != nil {
+			return err
+		}
+
+	case b.registry.HasOutputBinding(comp.Spec.Type, comp.Spec.Version):
+		if err := b.initOutputBinding(ctx, comp); err != nil {
+			return err
+		}
+
+	default:
+		diag.DefaultMonitoring.ComponentInitFailed(comp.Spec.Type, "creation", comp.ObjectMeta.Name)
+		return fmt.Errorf("couldn't find binding %s/%s", comp.Spec.Type, comp.Spec.Version)
+	}
+
+	return nil
+}
+
+func (b *binding) initInputBinding(ctx context.Context, comp compapi.Component) error {
+	if !b.isBindingOfDirection(bindingTypeInput, comp.Spec.Metadata) {
+		return nil
+	}
+
+	fName := comp.LogName()
+	binding, err := b.registry.CreateInputBinding(comp.Spec.Type, comp.Spec.Version, fName)
+	if err != nil {
+		diag.DefaultMonitoring.ComponentInitFailed(comp.Spec.Type, "creation", comp.ObjectMeta.Name)
+		return rterrors.NewInit(rterrors.CreateComponentFailure, fName, err)
+	}
+	err = binding.Init(ctx, bindings.Metadata{Base: b.meta.ToBaseMetadata(comp)})
+	if err != nil {
+		diag.DefaultMonitoring.ComponentInitFailed(comp.Spec.Type, "init", comp.ObjectMeta.Name)
+		return rterrors.NewInit(rterrors.InitComponentFailure, fName, err)
+	}
+
+	log.Infof("successful init for input binding %s (%s/%s)", comp.ObjectMeta.Name, comp.Spec.Type, comp.Spec.Version)
```

Comment:
In all these situations (also in other files), you can use `LogName()`:


```suggestion
	log.Infof("successful init for input binding %s", comp.LogName())
```